chore: Add instructions how to add a manuall PAT secret (#2383)
codenvy/che-docs: modules/end-user-guide/partials/proc_using-a-git-provider-access-token.adoc

.Procedure

. Copy your access token and encode it to Base64.
+
[source,terminal,subs="+quotes,+attributes"]
----
$ echo -n '__<your_access_token>__' | base64
----

. Prepare a new {orch-name} Secret in your user {orch-namespace}.
+
[source,yaml,subs="+quotes,+attributes,+macros"]
----
kind: Secret
apiVersion: v1
metadata:
  name: personal-access-token-__<your_chosen_name_for_this_token>__
  labels:
    app.kubernetes.io/component: scm-personal-access-token
    app.kubernetes.io/part-of: che.eclipse.org
  annotations:
    che.eclipse.org/che-userid: _<{prod-id-short}_user_ID>_ <1>
    che.eclipse.org/scm-personal-access-token-name: _<git_provider_name>_ <2>
    che.eclipse.org/scm-url: _<Git_provider_endpoint>_ <3>
    che.eclipse.org/scm-userid: _<Git_provider_user_ID>_ <4>
    che.eclipse.org/scm-username: _<Git_provider_username>_
data:
  token: __<Base64_encoded_access_token>__
type: Opaque
----
+
<1> Your Che user ID. You can retrieve `__<che-endpoint>__/api/user` to get the Che user data.
<2> The Git provider name: `github` or `gitlab` or `bitbucket-server`.
<3> The Git provider URL.

> <4> Your Git provider user ID.

Reviewer: @vinokurig, regarding `<4> Your Git provider user ID.`

I tried to look it up and `Consult the API-related sections of the Git provider's docs to get the user ID`, but no success.

Will you be able to clarify the following info?
I mean if you can just clarify it to me here in your reply/comment, and I can submit some proposed changes in response.

1.
[Which section and value in GitHub?](https://docs.github.com/en/rest/users)

2.
[Which subsection and value in GitLab?](https://docs.gitlab.com/ee/api/users.html)

3.
Which sections and values in BitBucket?
either
https://developer.atlassian.com/cloud/bitbucket/rest/api-group-users/#api-user-get; and in it, which exactly--`"uuid": "<string>"` under `Account` or ` "account_id": "<string>"` under `User`?
or
https://developer.atlassian.com/cloud/bitbucket/rest/api-group-users/#api-users-selected-user-get; and in it, which exactly--`"uuid": "<string>"` under `Account` or ` "account_id": "<string>"` under `User`?

Author: Fixed in the latest fixup commit

Reviewer: Thank you. I'm resolving this conversation.

If the administrator of your organization's {prod-short} instance has not configured xref:administration-guide:oauth-for-github-gitlab-or-bitbucket.adoc[OAuth for GitHub, GitLab, or Bitbucket], you can apply your personal access token as a {kubernetes} Secret.

Mounting your access token as a Secret enables the {prod-short} Server to access the remote repository that is cloned during workspace creation, including access to the repository's `/.che` and `/.vscode` folders.

Apply the {kubernetes} Secret in your user {orch-namespace} of the {orch-name} cluster of your organization's {prod-short} instance.

After you have applied the Secret, you can create new workspaces from a private GitHub, GitLab, or Bitbucket-server repository.

TIP: In your user {orch-namespace}, you can create and apply multiple access-token Secrets per a Git provider.

.Prerequisites

* An active `{orch-cli}` session, with administrative permissions, to the {orch-name} cluster. See {orch-cli-link}.

* The link:https://www.gnu.org/software/coreutils/base64[`base64`] command line tools are installed in the operating system you are using.

.Procedure

. Copy your access token and encode it to Base64.
+

> [source,terminal,subs="+quotes,+attributes"]

Reviewer 2: ```suggestion
[subs="+quotes,+attributes"]
```

Author: I've found several places where [source,terminal,  is used, so I think we should merge this as it is and consider to rework all the findings in another PR.

----
$ echo -n '__<your_access_token>__' | base64
----

. Prepare a new {orch-name} Secret in your user {orch-namespace}.
+
[source,yaml,subs="+quotes,+attributes,+macros"]
----
kind: Secret
apiVersion: v1
metadata:
  name: personal-access-token-__<your_chosen_name_for_this_token>__
  labels:
    app.kubernetes.io/component: scm-personal-access-token
    app.kubernetes.io/part-of: che.eclipse.org
  annotations:
    che.eclipse.org/che-userid: _<{prod-id-short}_user_ID>_ <1>
    che.eclipse.org/scm-personal-access-token-name: _<git_provider_name>_ <2>
    che.eclipse.org/scm-url: _<Git_provider_endpoint>_ <3>
    che.eclipse.org/scm-userid: _<Git_provider_user_ID>_ <4>
    che.eclipse.org/scm-username: _<Git_provider_username>_
data:
  token: __<Base64_encoded_access_token>__
type: Opaque
----
+
<1> Your Che user ID. You can retrieve `__<che-endpoint>__/api/user` to get the Che user data.
<2> The Git provider name: `github` or `gitlab` or `bitbucket-server`.
<3> The Git provider URL.

> <4> Your Git provider user ID, follow the API documentation to retrieve the user object:
> * GitHub: https://docs.github.com/en/rest/users/users#get-a-user.
> * GitLab: https://docs.gitlab.com/ee/api/users.html#for-normal-users.
> * Bitbucket Server: https://developer.atlassian.com/server/bitbucket/rest/v802/api-group-users/#api-users-userslug-get. Get the `account_id` value.

Reviewer: ```suggestion
<4> Your Git provider user ID. Follow the API documentation to retrieve the user object:
* GitHub: link:https://docs.github.com/en/rest/users/users#get-a-user[Get a user].
* GitLab: link:https://docs.gitlab.com/ee/api/users.html#for-normal-users[List users: For normal users].
* Bitbucket Server: link:https://developer.atlassian.com/server/bitbucket/rest/v802/api-group-users/#api-users-userslug-get[Get users {userSlug}]. Find the `account_id` value within the response to the `GET /rest/api/1.0/users/{userSlug}` request.
```

Reviewer: ![image](https://user-images.githubusercontent.com/78564304/180790243-a939de38-cbb1-4c4a-b493-a16c520f2575.png)

Reviewer: BTW, I can't find the `account_id` value in the Bitbucket Server docs (https://developer.atlassian.com/server/bitbucket/rest/v802/api-group-users/#api-users-userslug-get). Could you please double-check that? Is it `id`? Or where is it exactly? If it's not explicitly mentioned in the linked docs, then which documented procedure in the linked docs gets it?

Author: > List users: For normal users

The idea is to get user by specific `username`, the request example is listed below, so I think it should be named like: `Get a user`.
> I can't find the account_id value in the Bitbucket Server docs

I don't know why the documentation page shows such example, but the real request returns such JSON:
```JSON
{
  "username": "ivinokur",
  "has_2fa_enabled": null,
  "display_name": "ivinokur",
  "account_id": "id-string",
  "links": {
    "hooks": {
      "href": "https://api.bitbucket.org/2.0/workspaces/%7Be3666240-4b13-4d3e-928f-32d5692c8aa3%7D/hooks"
    },
    "self": {
      "href": "https://api.bitbucket.org/2.0/users/%7Be3666240-4b13-4d3e-928f-32d5692c8aa3%7D"
    },
    "repositories": {
      "href": "https://api.bitbucket.org/2.0/repositories/%7Be3666240-4b13-4d3e-928f-32d5692c8aa3%7D"
    },
    "html": {
      "href": "https://bitbucket.org/%7Be3666240-4b13-4d3e-928f-32d5692c8aa3%7D/"
    },
    "avatar": {
      "href": "https://secure.gravatar.com/avatar/f583ff5c1e2669bc67f395b6c8c8c046?d=https%3A%2F%2Favatar-management--avatars.us-west-2.prod.public.atl-paas.net%2Finitials%2FI-2.png"
    },
    "snippets": {
      "href": "https://api.bitbucket.org/2.0/snippets/%7Be3666240-4b13-4d3e-928f-32d5692c8aa3%7D"
    }
  },
  "nickname": "ivinokur",
  "created_on": "2016-06-29T08:53:43.616762+00:00",
  "is_staff": false,
  "location": null,
  "account_status": "active",
  "type": "user",
  "uuid": "{string}"
}
```

Reviewer: Thanks for the clarification.

> The idea is to get user by specific username, the request example is listed below, so I think it should be named like: Get a user.

I use the heading that appears in the linked docs to help the reader find the correct section there. By the way, what should the user look for in the request response from GitLab? `"id"`?

> I don't know why the documentation page shows such example, but the real request returns such JSON:

Great. Thanks. I updated my suggestion above. Please see above.

Looking at the linked GitHub docs again, what's the specific parameter that the reader should be looking for in the request response from GitHub? We can add it here too.

Author: Here is the updatet version:
![screenshot-0 0 0 0_4000-2022 07 26-12_02_20](https://user-images.githubusercontent.com/7668752/180968050-31a6b85a-e1d6-46e2-92f5-4066c18f60fc.png)

Reviewer: Thank you. I'm resolving this conversation.

> * GitHub: link:https://docs.github.com/en/rest/users/users#get-a-user[Get a user]. Find the `id` value.
> * GitLab: link:https://docs.gitlab.com/ee/api/users.html#for-normal-users[List users: For normal users], use the `username` filter: `/users?username=:username`. Find the `id` value.
> * Bitbucket Server: link:https://developer.atlassian.com/server/bitbucket/rest/v802/api-group-users/#api-users-userslug-get[Get users {userSlug}]. Find the `account_id` value.

Reviewer: I wonder if we can clarify where the reader should look for the values (not in the linked page, but in the API response)?
```suggestion
* GitHub: link:https://docs.github.com/en/rest/users/users#get-a-user[Get a user]. See the `id` value in the response.
* GitLab: link:https://docs.gitlab.com/ee/api/users.html#for-normal-users[List users: For normal users], use the `username` filter: `/users?username=:username`. See the `id` value in the response.
* Bitbucket Server: link:https://developer.atlassian.com/server/bitbucket/rest/v802/api-group-users/#api-users-userslug-get[Get users {userSlug}]. See the `account_id` value in the response.
```

> * Bitbucket Server: link:https://developer.atlassian.com/server/bitbucket/rest/v802/api-group-users/#api-users-userslug-get[Get users {userSlug}]. See the `account_id` value in the response.

Reviewer: Sorry about this. I should have thought about this earlier!
```suggestion
* Bitbucket Server: link:https://developer.atlassian.com/server/bitbucket/rest/v802/api-group-users/#api-users-userslug-get[Get users]. See the `account_id` value in the response.
```

. Apply the Secret.
+

> [source,terminal,subs="+quotes,+attributes"]

Reviewer 2: ```suggestion
[subs="+quotes,+attributes"]
```

Author: I've found several places where [source,terminal,  is used, so I think we should merge this as it is and consider to rework all the findings in another PR.
